jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java

```diff
@@ -247,21 +247,25 @@ private void addStepsWithLifecycle(AbstractPerformableScenario performableScenar
 		performableScenario.addBeforeSteps(context.lifecycleSteps(lifecycle, storyAndScenarioMeta, Stage.BEFORE));
 		addMetaParameters(parameters, storyAndScenarioMeta);
 		performableScenario.addGivenStories(performableGivenStories(context, scenario.getGivenStories(),
-		        parameters));
+		        parameters, storyAndScenarioMeta));
 		performableScenario.addSteps(context.scenarioSteps(scenario, parameters));
 		performableScenario.addAfterSteps(context.lifecycleSteps(lifecycle, storyAndScenarioMeta, Stage.AFTER));
 		performableScenario.addAfterSteps(context.beforeOrAfterScenarioSteps(storyAndScenarioMeta, Stage.AFTER,
                 ScenarioType.ANY));
 	}
 
-    private List<PerformableStory> performableGivenStories(RunContext context, GivenStories givenStories,
-            Map<String, String> parameters) {
+    private List<PerformableStory> performableGivenStories(RunContext context, GivenStories givenStories, Map<String,
+            String> parameters, Meta meta) {
         List<PerformableStory> stories = new ArrayList<PerformableStory>();
         if (givenStories.getPaths().size() > 0) {
             for (GivenStory givenStory : givenStories.getStories()) {
                 RunContext childContext = context.childContextFor(givenStory);
                 // run given story, using any parameters provided
-                Story story = storyOfPath(context.configuration(), childContext.path());                
+                Story story = storyOfPath(context.configuration(), childContext.path());
+                if (!meta.isEmpty()) {
+                    story = new Story(story.getPath(), story.getDescription(), story.getMeta().inheritFrom(meta)
+                            , story.getNarrative(), story.getScenarios());
```

Comment:
The same constructor is used later in the method storyWithMatchingScenarios at [https://github.com/ibachyla/jbehave-core/blob/07a49f4dc49a727e0f2056c28d3dec68dcda640d/jbehave-core/src/main/java/org/jbehave/core/embedder/PerformableTree.java#L270](url)